homeassistant/components/conversation/__init__.py

```diff
@@ -297,6 +298,45 @@ async def websocket_list_agents(
     connection.send_message(websocket_api.result_message(msg["id"], {"agents": agents}))
 
 
+@websocket_api.websocket_command(
+    {
+        vol.Required("type"): "conversation/agent/homeassistant/debug",
+        vol.Required("sentences"): [str],
+        vol.Optional("language"): str,
+        vol.Optional("device_id"): vol.Any(str, None),
+    }
+)
+@websocket_api.async_response
+async def websocket_hass_agent_debug(
+    hass: HomeAssistant, connection: websocket_api.ActiveConnection, msg: dict
+) -> None:
+    """Return intents that would be matched by the default agent for a list of sentences."""
+    agent = await _get_agent_manager(hass).async_get_agent(HOME_ASSISTANT_AGENT)
+    assert isinstance(agent, DefaultAgent)
+    results = [
+        await agent.async_recognize(
+            ConversationInput(
+                text=sentence,
+                context=connection.context(msg),
+                conversation_id=None,
+                device_id=msg.get("device_id"),
+                language=msg.get("language", hass.config.language),
+            )
+        )
+        for sentence in msg["sentences"]
+    ]
+
+    # Return results for each sentence in the same order as the input.
+    connection.send_result(
```

Comment:
Although the test has been updated to no longer show it, we're still returning too much information here. Users are going to paste 100 sentences in here, and we're basically going to return 100x all sentences that match HassTurnOn or something. Exploring the registered intents are out of scope for this endpoint.

Reply:
I would also suggest that we go back to comparing the whole result against a `snapshot` so it's clear what we are sending over the wire. Also allows us to catch if we're accidentally send too much in the future.

Reply:
OK, I've switched to just the matching intent name and entities (name, value, and actual match text).